### PR TITLE
Move components in App.js to their own individual files

### PR DESCRIPTION
**Pull request by colefortner**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This PR separates NavLinks, AddItem and ItemList components into their own files.

## Related Issue

Refactors code from https://github.com/the-collab-lab/tcl-41-smart-shopping-list/issues/2

## Acceptance Criteria

 - [x] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- [x] When one of the links is clicked, the browser URL updates to represent the current view, and a matching view component is displayed
- [x] Whichever view is selected, the corresponding link should display in bold text
- [x] File structure and file/component/route names are agreeable to the tcl-41 team and mentors.  
- [x] No changes should be observed in app appearance or functionality.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :hammer: Refactoring       |

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin jc-cf-separate-components` and check that branch out with `git checkout jc-cf-separate-components`
* Then use `npm install` to install the newly added dependency locally and `npm start` to launch the app.
* Check that the links are present at the bottom of the page. Click the links and make sure the view text matches the URL; the home page should say "List of Items" and the /add-an-item page should say "Add an Item". Also, check that the link for the active page is bold.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,46 +1,11 @@
 import React from 'react';
-import { Routes, Route, NavLink } from 'react-router-dom';
+import { Routes, Route } from 'react-router-dom';
+
 import './App.css';
+import NavLinks from './Navigation/NavLinks';
```
Comment (jmc617) [inline src/App.js:5]: Good idea creating a separate folder for navigation components!

File: src/App.js [hunk 2]
```diff
@@ -50,8 +15,8 @@ function App() {
       </header>
       <h1>Welcome to your Smart Shopping List!</h1>
       <Routes>
-        <Route path="/" element={<List />} />
-        <Route path="add-an-item" element={<AddAnItem />} />
+        <Route path="/item-list" element={<ItemList />} />
+        <Route path="/add-item" element={<AddItem />} />
```
Comment (jmc617) [inline src/App.js:19]: I like the renaming!

**Review by jmc617 — COMMENTED** [review]
When I tested the code it worked exactly as before the changes were made. Thanks for doing this Cole!

**Review by MikeBLambert — APPROVED** [review]
Looks good!

**Pull request closed without merging**